### PR TITLE
Add support for generic STM32G030F6

### PR DESCRIPTION
**Pull request by ademuri**
This adds support for generic `STM32G030F6`-based boards. I have a custom board which uses this processor. I partially tested the `CheckVariant` sketch - I verified that the LED blinks slowly, but did not check the serial output.

### CODE REVIEW

**Comment by fpistm** [comment]
Hi @ademuri 
Thanks for the PR, @ABOSTM will review it.
As a first feedback, I would recommend to update the generic clock config.

**Review by ABOSTM — CHANGES_REQUESTED** [review]
@ademuri
Thanks for this Pull Request. As said @fpistm, it would be better to redefine the generic clock config to benefit from better performances
https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-(board)#3---generic-system-clock-configuration
otherwise you stuck on default reset clock (which is slow).

File: README.md [hunk 1]
```diff
@@ -294,6 +294,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | Status | Device(s) | Name | Release | Notes |
 | :----: | :-------: | ---- | :-----: | :---- |
 | :yellow_heart: | STM32G030C6<br>STM32G030C8 | Generic Board | **2.2.0** |
+| :green_heart: | STM32G030F6<br>STM32G030F6 | Generic Board | **2.2.0** |
```
[the remark this answers is not in the document]
Comment (ademuri) [reply README.md]: Done.

**Comment by ademuri** [comment]
Updated the clock.

File: variants/STM32G0xx/G030F6P/generic_clock.c [hunk 2]
```diff
@@ -20,8 +20,41 @@
   */
 WEAK void SystemClock_Config(void)
 {
-  /* SystemClock_Config can be generated by STM32CubeMX */
-#warning "SystemClock_Config() is empty. Default clock at reset is used."
+    RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+  RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
```
Comment (ABOSTM) [inline variants/STM32G0xx/G030F6P/generic_clock.c:24]: With compiler we used, assignation to 0 on such heterogenous structure causes compiler warning, 
(and are not required) 
```suggestion
    RCC_OscInitTypeDef RCC_OscInitStruct = {};
  RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
```

**Review by ABOSTM — CHANGES_REQUESTED** [review]
Almost good.
See inline comment.
Also could you run Astyle to fix CI issue:
https://github.com/stm32duino/wiki/wiki/Astyle

File: variants/STM32G0xx/G030F6P/generic_clock.c [hunk 3]
```diff
@@ -20,8 +20,41 @@
...
+  RCC_OscInitStruct.HSIDiv = RCC_HSI_DIV1;
+  RCC_OscInitStruct.HSICalibrationValue = RCC_HSICALIBRATION_DEFAULT;
+  RCC_OscInitStruct.PLL.PLLState = RCC_PLL_ON;
+  RCC_OscInitStruct.PLL.PLLSource = RCC_PLLSOURCE_HSI;
+  RCC_OscInitStruct.PLL.PLLM = RCC_PLLM_DIV1;
+  RCC_OscInitStruct.PLL.PLLN = 8;
+  RCC_OscInitStruct.PLL.PLLP = RCC_PLLP_DIV2;
+  RCC_OscInitStruct.PLL.PLLR = RCC_PLLR_DIV2;
+  if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK)
+  {
```
Comment (fpistm) [inline variants/STM32G0xx/G030F6P/generic_clock.c:43]:
```suggestion
  if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK) {
```

File: variants/STM32G0xx/G030F6P/generic_clock.c [hunk 4]
```diff
@@ -20,8 +20,41 @@
...
+  /** Initializes the CPU, AHB and APB buses clocks
+  */
+  RCC_ClkInitStruct.ClockType = RCC_CLOCKTYPE_HCLK|RCC_CLOCKTYPE_SYSCLK
+                              |RCC_CLOCKTYPE_PCLK1;
+  RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_PLLCLK;
+  RCC_ClkInitStruct.AHBCLKDivider = RCC_SYSCLK_DIV1;
+  RCC_ClkInitStruct.APB1CLKDivider = RCC_HCLK_DIV1;
+
+  if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_2) != HAL_OK)
+  {
```
Comment (fpistm) [inline variants/STM32G0xx/G030F6P/generic_clock.c:55]:
```suggestion
  if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK) {
    Error_Handler();
  }
  /** Initializes the CPU, AHB and APB buses clocks
  */
  RCC_ClkInitStruct.ClockType = RCC_CLOCKTYPE_HCLK | RCC_CLOCKTYPE_SYSCLK
                                                 | RCC_CLOCKTYPE_PCLK1;
  RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_PLLCLK;
  RCC_ClkInitStruct.AHBCLKDivider = RCC_SYSCLK_DIV1;
  RCC_ClkInitStruct.APB1CLKDivider = RCC_HCLK_DIV1;

  if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_2) != HAL_OK) {
```

**Review by ABOSTM — APPROVED** [review]
LGTM

**Comment by vik0asth** [comment]
Thanks for adding the support for this MCU in latest version. I have a custom board with STM32G0F6Px MCU also. I am trying to run tone example on that but not getting any output/ sound on buzzer.

Same tone example code runs fine on Nucleo-64 STM32G071RB eval board but not on custom board. The "variant_generic.h" files for the two MCUs uses two different timers - could that be the problem? TIM6/TIM7 are not supported on STM32G0F6Px, so which timer to use?

image

Can anyone please help to resolve this issue?

Thanks in advance.

**Pull request merged**